sdk/src/auth/composable.ts

```diff
@@ -119,36 +124,40 @@ export const authentication = (mode: AuthenticationMode = 'cookie', config: Part
 			return refreshPromise;
 		};
 
-		return {
-			refresh,
-			async login(email: string, password: string, options: LoginOptions = {}) {
-				await resetStorage();
+		function login(payload: LocalLoginPayload, options?: LoginOptions): Promise<AuthenticationData>;
+		function login(payload: LDAPLoginPayload, options?: LoginOptions): Promise<AuthenticationData>;
+		async function login(payload: LoginPayload, options: LoginOptions = {}) {
+			await resetStorage();
```

Comment:
The following should allow for backward compatibility. The legacy login method is at the back so it shows up last for code completion.

```suggestion
		function login(payload: LocalLoginPayload, options?: LoginOptions): Promise<AuthenticationData>;
		function login(payload: LDAPLoginPayload, options?: LoginOptions): Promise<AuthenticationData>;
		function login(email: string, password: string, options?: LoginOptions): Promise<AuthenticationData>;
		async function login(
			emailOrPayload: string | LoginPayload | LDAPLoginPayload,
			passwordOrOptions?: string | LoginOptions,
			maybeOptions?: LoginOptions,
		): Promise<AuthenticationData> {
			// Support legacy login method with email and password
			if (typeof emailOrPayload === 'string' && typeof passwordOrOptions === 'string') {
				const email = emailOrPayload;
				const password = passwordOrOptions;
				const options = maybeOptions || {};

				return login({ email, password }, options);
			}

			const payload = emailOrPayload as LocalLoginPayload | LDAPLoginPayload;
			const options = passwordOrOptions as LoginOptions;

			await resetStorage();
```